processor/lsmintervalprocessor/processor_test.go

```diff
@@ -118,3 +120,90 @@ func TestAggregation(t *testing.T) {
 		})
 	}
 }
+
+// BenchmarkAggregation benchmarks various inputs with high resource attribute cardinality, and with and without OTTL.
+func BenchmarkAggregation(b *testing.B) {
+	testCases := []struct {
+		name        string
+		passThrough bool
+	}{
+		{name: "sum_cumulative"},
+		{name: "sum_delta"},
+		{name: "histogram_cumulative"},
+		{name: "histogram_delta"},
+		{name: "exphistogram_cumulative"},
+		{name: "exphistogram_delta"},
+		{name: "summary_enabled"},
+		{name: "summary_passthrough", passThrough: true},
+	}
+
+	ctx, cancel := context.WithCancel(context.Background())
+	defer cancel()
+
+	for _, ottlStatements := range [][]string{
+		{},
+		{
+			`set(resource.attributes["custom_res_attr"], "res")`,
+			`set(instrumentation_scope.attributes["custom_scope_attr"], "scope")`,
+			`set(attributes["custom_dp_attr"], "dp")`,
+			`set(resource.attributes["dependent_attr"], Concat([attributes["aaa"], "dependent"], "-"))`,
+		},
```

Comment:
_nit_: I would rather split the benchmarks into 2, like `BenchmarkAggregation` and `BenchmarkAggregationWithOTTL` and refactor the test to use a helper method that would run all the test cases with the ottl statements as an input.

Reply:
good call!